### PR TITLE
CSM-239 Change transaction block representation

### PR DESCRIPTION
**Pull request by ksaric**
https://issues.serokell.io/issue/CSM-239

And added a few comments, refactored a bit of code.

### CODE REVIEW

File: src/Pos/Explorer/Web/Server.hs [hunk 1]
```diff
@@ -407,29 +435,32 @@ topsortTxsOrFail f =
...
-    MM.lookupM EX.getTxExtra id =<< getTxExtra
--}
-getTxExtraOrFail :: ExplorerMode m => TxId -> m TxExtra
-getTxExtraOrFail id =
-    maybeThrow (Internal "Transaction not found") =<< getTxExtra id
+
+-- | Get transaction extra from the database, and if you don't find it
+-- throw an exception.
+getTxExtraOrFail :: MonadDB m => TxId -> m TxExtra
+getTxExtraOrFail id = getTxExtra id >>= maybeThrow exception
```
Comment (akegalj) [inline src/Pos/Explorer/Web/Server.hs]: Use more descriptive name then `id`. Id is usually identify function and looks confusing at first. `txId` would do

Comment (akegalj) [inline src/Pos/Explorer/Web/Server.hs]: Also, you can do it point free notation
```
maybeThrow exception <=< getTxExtra
```

and lose `id`

File: src/Pos/Explorer/Web/Server.hs [hunk 2]
```diff
@@ -407,29 +435,32 @@ topsortTxsOrFail f =
     maybeThrow (Internal "Dependency loop in txs set") .
     topsortTxs f
 
+-- | Convert server address to client address,
+-- with the possible exception (effect).
 cAddrToAddr :: MonadThrow m => CAddress -> m Address
-cAddrToAddr cAddr =
-    fromCAddress cAddr &
-    either (const $ throwM $ Internal "Invalid address!") pure
+cAddrToAddr cAddr = either exception pure (fromCAddress cAddr)
```
Comment (akegalj) [inline src/Pos/Explorer/Web/Server.hs]: `either exception pure . fromCAddress`

File: src/Pos/Explorer/Web/Server.hs [hunk 3]
```diff
@@ -407,29 +435,32 @@ topsortTxsOrFail f =
...
+  where
+    exception = const $ throwM $ Internal "Invalid address!"
 
+-- | Convert server transaction to client transaction,
+-- with the possible exception (effect).
 cTxIdToTxId :: MonadThrow m => CTxId -> m TxId
-cTxIdToTxId cTxId =
-    fromCTxId cTxId &
-    either (const $ throwM $ Internal "Invalid transaction id!") pure
+cTxIdToTxId cTxId = either exception pure (fromCTxId cTxId)
```
Comment (akegalj) [inline src/Pos/Explorer/Web/Server.hs]: `either exception pure . fromCTxId`

File: src/Pos/Explorer/Web/Server.hs (hunk 1 above)
Comment (ksaric) [inline src/Pos/Explorer/Web/Server.hs]: Rename `id` is indeed a good idea. Will fix.

File: src/Pos/Explorer/Web/Server.hs (hunk 2 above)
Comment (ksaric) [inline src/Pos/Explorer/Web/Server.hs]: I dislike point-free. What I wrote is explicit, easier to read and understand. I don't have to look at the types to know what I'm doing.

Comment (akegalj) [inline src/Pos/Explorer/Web/Server.hs]: @gromakovsky which style should we prefer in this use case ?

Comment (gromakovsky) [inline src/Pos/Explorer/Web/Server.hs]: [Style guide](https://github.com/serokell/serokell-util/blob/master/serokell-style.md#point-free-style) only suggests avoiding over-using point-free style (which is not the case with `either exception pure . fromCAddress`). I think both are acceptable here depending on author's preferences.

**Pull request merged**